### ISSUE TITLE
Replace static references to feed height and width with non-static object references

### ISSUE THREAD

**Issue by cbdmaul** [phrack/ShootOFF#287]
We use static references to CameraManager.FEED_HEIGHT and FEED_WIDTH throughout the code.  These should be changed to references to the correct CameraManager's object members instead.  This will take some changes as the CameraManager is not passed to all the classes that use those static references.

This is to support future changes to ShootOFF.

**Comment by cbdmaul** [comment]
1. Dedupe processor needs to be on a per-camera basis.  It is currently assigned to the config.  Isn't this a bug anyway? If you have two shots at the same place on two feeds, they'll be picked up as a dupe.
2. The GUI elements that reference the feed height and width for display purposes need to use some different global default, preferably user configurable, for how large to show it.  I would appreciate this as 640x480 is frickin tiny on a 4k display.  It follows that the target bounds would use this instead of the feed settings.
3. Rolling recorder should have its video settings separated from the feed settings.
4. Autocalibration should be given bounds

**Comment by phrack** [comment]
Deduplicators should probably be static in the CanvasManager and be used at or towards the top of addShot. Agreed it's a bug for it to be universal and even in the Configuration at all.

**Comment by cbdmaul** [comment]
I pushed a new branch, code in there now fixes the 1st problem.  I don't see that as a big enough bug to merge into master but I'll leave that to you.  Wasn't that much code and it passes tests.

Edit: Although I hadn't seen you comment and I solved it a very slightly different way.

My logic is that the cameramanager is more tightly tied to it, the dedupe needs to know the res and the shotdetection needs the dedupe.

**Comment by phrack** [comment]
I am going to open a separate issue for the dedupe bug to make it so I can merge it early. Since this is a bug fix it fits within the current "bug fixes only" code freeze I have in place.

**Comment by cbdmaul** [comment]
I pushed a new commit that makes it static and final in CameraManager.  Tests are passing.

**Comment by phrack** [comment]
Is it mergable to close #300?

**Comment by cbdmaul** [comment]
No, hold on, there's obviously something wrong with the commit if you look at the diff.

**Comment by phrack** [comment]
I'll stand by until you say it's ready. I don't usually bother to look at the diff unless I am told the code is done.

**Comment by cbdmaul** [comment]
Newest commit f8e11ed4da76125daa62788a57cc9d943fe823cf handles the outstanding issues and tests pass.  However, no other resolutions have been tested yet which needs to be done before this is ready for prime time.  There will undoubtedly be translation issues if the display res and feed res differ which has not yet been handled.

**Comment by phrack** [comment]
Closed by pulll request #319
